### PR TITLE
[#455]  Add institution selection into super Admin and Admin settings page

### PR DESCRIPTION
**Pull request by danielkryska**
- [x] On input show filtered institutions
- [x] Keep chosen institution in memory/local storage/storage
- [x] Add input to the sidebar at `settings/profile` page for super admin (at the top)
- [x] Add input to `Zarzadaj instytucjami` tile at the main page of `settings/dashboard` for super Admin
- [x] On select/cache add queryParam to url and keep it in there

Generic search results input
- [x] Create generic search input
- [x] Replace locations search with generic one (input on `/map/products` sidebar)
- [x] Reparation of select first resut method (choose first result by clicking on the loop icon)

### CODE REVIEW

**Comment by jswk** [comment]
When I enter the search term for the first time after reload, the input is cleared after a moment. When the results box is shown.
I guess it should retain the input, not discard it on loading the results.

File: s4e-web/src/app/views/settings/admin-dashboard/admin-dashboard.component.html [hunk 1]
```diff
@@ -8,14 +8,36 @@ <h1 dashboard-title>Witamy w panelu administratora Sat4Envi</h1>
     <s4e-tile>
       <header class="panel__header" tile-title>Wybierz instytucję</header>
       <p i18n tile-description>Wybierz instytucję aby móc nią zarządzać</p>
-      <input class="search__input" placeholder="Wpisz szukaną instytucję..."
-        i18n-placeholder footer-navigation />
+      <s4e-search
+        placeholder="Wpisz szukany instytut..."
```
Comment (jswk) [inline s4e-web/src/app/views/settings/admin-dashboard/admin-dashboard.component.html:12]: instytut -> instytucję

File: s4e-web/src/app/views/settings/settings.component.html [hunk 2]
```diff
@@ -1,23 +1,85 @@
...
+          i18n
+        >
+          Dashboard
+        </a>
       </li>
+    </ul>
+    <ul class="navigation__list institution__actions">
+      <li *ngIf="showInstitutions$ | async">
+        <s4e-search
+          placeholder="Wpisz szukany instytut..."
```
Comment (jswk) [inline s4e-web/src/app/views/settings/settings.component.html:17]: instytut -> instytucję

File: s4e-web/src/app/views/settings/admin-dashboard/admin-dashboard.component.html [hunk 3]
```diff
@@ -8,14 +8,36 @@ <h1 dashboard-title>Witamy w panelu administratora Sat4Envi</h1>
...
+        [searchResults]="institutions$ | async"
+        [isSearchLoading]="institutionsLoading$ | async"
+        [isSearchOpen]="isInUse && areResultsOpen$ | async"
+        [value]="searchValue"
+
+        (selectResult)="selectInstitution($event)"
+        (refreshResults)="searchForInstitutes($event)"
+        (selectFirstResult)="selectFirstInstitute()"
```
Comment (jswk) [inline s4e-web/src/app/views/settings/admin-dashboard/admin-dashboard.component.html:20]: Again, naming. Institute->Institution. Look for other places with that too

**Comment by danielkryska** [comment]
The styles of elements was improved, but in the case of missing `institutions` bug, I'm not able to find.

search 'zarz'

**Comment by jswk** [comment]
> The styles of elements was improved, but in the case of missing `institutions` bug, I'm not able to find.

Thanks for style improvement. The element floats correctly now. Also, the search identifies the missing institution now, I don't know what was the issue.

There is a slight problem with naming, when you type into the institution search box, you don't actually search by prefix, as naming your arguments and variables suggests, but rather by "contains" relation.

You also send two requests to endpoint `/institutions`institutePrefix, is it by design?
image

There is also an issue with search:
1. Type "zarz" into the search box, the result will appear.
2. Cancel the search by clicking the X.
3. The result for "zarz" re-appears.

^^
This was checked on 9229cfb0c960d313296a7ce8fd011e705a02d21e.

**Comment by danielkryska** [comment]
> There is a slight problem with naming, when you type into the institution search box, you don't actually search by prefix, as naming your arguments and variables suggests, but rather by "contains" relation.
**Done**

> You also send two requests to endpoint /institutionsinstitutePrefix, is it by design?
It was case of one additional line. **Done**

> There is also an issue with search:

>    Type "zarz" into the search box, the result will appear.
>    Cancel the search by clicking the X.
>    The result for "zarz" re-appears.
I can't replicate this issue. Have you done any additional step?

**Comment by jswk** [comment]
> > Type "zarz" into the search box, the result will appear.
> > Cancel the search by clicking the X.
> > The result for "zarz" re-appears.
> 
> I can't replicate this issue. Have you done any additional step?

I don't think so. I do this on https://pr-463-sok.docker-fid.grid.cyf-kr.edu.pl.
I clear the site storage and proceed to the settings directly after login to `zkAdmin@mail.pl`.
out

**Comment by danielkryska** [comment]
Ok @jswk . Now I understand what I see. It's a feature not a bug :P In other words, we'll have only few institutions, in that case focusing on input (including cleaning) show you up all possible institutions you can choose. Although as you do not feel it, maybe it should be removed...

**Pull request merged**